### PR TITLE
[osx] - fix refclock after refreshrate change and re-add videoscreen.delayrefreshchange

### PR DESCRIPTION
**Pull request by Memphiz**
This fixes a problem where after changing refreshrate the VideoPlayer still uses the old fps. It also re-adds the videoscreen.delayrefreshchange which got lost somewhere during the cycle (used the same pattern as on WindowingX11.

@FernetMenta so far what we gathered i guess. Fine with this?

Also the user which reported the choppy vtb (which you fixed with the fence logic inversion) stated that its smooth now - but it gets choppy as soon as an overlay comes into play :(. So this issue is still left then.

### CODE REVIEW

**Comment by FernetMenta** [comment]
great! big step forward.
btw: the pr shows the label "Backport-Done"

**Comment by Memphiz** [comment]
It shows the label backport done because i have the backport pr done too. Thought that was how it was meant to be used?

**Comment by FernetMenta** [comment]
maybe, not sure. backport-done somehow implies that it hit the code already.

**Comment by Memphiz** [comment]
Oh i always use it as - pull request has been planted.

**Comment by MartijnKaijser** [comment]
any hold up for merge (and the backport)?

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
fails on windows/osx

File: xbmc/windowing/osx/WinSystemOSX.mm [hunk 1]
```diff
@@ -1742,8 +1743,33 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     (*i)->OnLostDisplay();
 }
 
+void CWinSystemOSX::HandleOnResetDevice()
+{
+  
+  int delay = CSettings::GetInstance().GetInt("videoscreen.delayrefreshchange");
```
Comment (FernetMenta) [inline xbmc/windowing/osx/WinSystemOSX.mm]: CServiceBroker::GetSettings()

**Comment by Memphiz** [comment]
yep PR updated - sorry for that one - jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**